ui/panels-plugin/src/plugins/time-series-chart/utils/palette-gen.ts

```diff
@@ -11,39 +11,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { PaletteOptions } from '../time-series-chart-model';
+import ColorHash from 'color-hash';
 
 /**
- * Helper function to generate a random color for a chart series based on its name
+ * Get color from generative color palette
  */
-export function getRandomColor(identifier: string): string {
-  let hash = 0;
-  for (let index = 0; index < identifier.length; index++) {
-    hash = identifier.charCodeAt(index) + ((hash << 5) - hash);
-  }
-  // Use HSLA to only get random "bright" colors from this
-  const color = `hsla(${~~(180 * hash)},50%,50%,0.8)`;
-  return color;
+export function getAutoPaletteColor(name: string, fallbackColor: string): string {
+  // corresponds to 'Auto' in palette.kind for generative color palette
+  const generatedColor = getConsistentSeriesNameColor(name);
+  return generatedColor ?? fallbackColor;
 }
 
 /**
  * Get line color as well as color for tooltip and legend, account for whether palette is 'Cateogrical' or 'Auto' (generative)
  */
-export function getSeriesColor(
-  name: string,
-  seriesIndex: number,
-  palette: string[],
-  paletteKind: PaletteOptions['kind'] = 'Auto'
-): string {
-  if (paletteKind === 'Categorical' && Array.isArray(palette)) {
-    const colorIndex = seriesIndex % palette.length;
-    // TODO: take fallback color from theme
-    const seriesColor = palette[colorIndex];
-    if (seriesColor !== undefined) {
-      return seriesColor;
-    }
-  }
-
-  // corresponds to 'Auto' in palette.kind
-  return getRandomColor(name);
+export function getCategoricalPaletteColor(palette: string[], paletteIndex: number, fallbackColor: string): string {
+  // Loop through predefined static color palette
+  const colorIndex = paletteIndex % palette.length;
```

Comment:
I like the idea where we reduce the opacity if it loops again, what do you think?

Reply:
Update: I'm thinking I'll merge this PR first and then follow up with the new approach in this PR #1124 based on Mohit's feedback. In that PR, the palette is automatically changed based on total series so it wouldn't ever need to loop through the new default Categorical palette a second time

Reply:
This makes sense to me! I'm going to go ahead and merge this PR since CI is green and update variables based on this feedback in #1124 (I'm touching the getCategoricalPaletteColor function there too)